### PR TITLE
Fix pluralize and singularize for UpperCamelCase case.

### PR DESCRIPTION
**Pull request by hikouki**
Hi! 🍣 

Is this bug?

I am using Laravel.
`TargetHuman` string was replaced to  `TargetHumen` string by doctrine/inflector.

### CODE REVIEW

File: lib/Doctrine/Common/Inflector/Inflector.php [hunk 1]
```diff
@@ -402,10 +404,13 @@ public static function pluralize($word)
             self::$plural['cacheIrregular']   = '(?:' . implode('|', array_keys(self::$plural['merged']['irregular'])) . ')';
         }
 
-        if (preg_match('/(.*)\\b(' . self::$plural['cacheIrregular'] . ')$/i', $word, $regs)) {
-            self::$cache['pluralize'][$word] = $regs[1] . substr($word, 0, 1) . substr(self::$plural['merged']['irregular'][strtolower($regs[2])], 1);
-            
-            return self::$cache['pluralize'][$word];
+        if (preg_match('/(.*(?:\\b|_))(' . self::$plural['cacheIrregular'] . ')$/i', $tablized_word, $regs)) {
+            $result = $regs[1] . substr($regs[2], 0, 1) . substr(self::$plural['merged']['irregular'][strtolower($regs[2])], 1);
+            if ($tablized_word !== $word) {
```
Comment (Ocramius) [inline lib/Doctrine/Common/Inflector/Inflector.php]: Naming is a bit weird - is this the ubiquitous language here? I really have no idea myself :-\

**Comment by jwage** [comment]
Hi @hikouki, thanks for the contribution. Do you have any availability to continue work on this PR? I would like to include it in an upcoming release potentially.

**Comment by hikouki** [comment]
OK, I will try. ✈️

**Comment by hikouki** [comment]
I tried. Is this OK?

File: lib/Doctrine/Common/Inflector/Inflector.php [hunk 2]
```diff
@@ -486,6 +486,8 @@ public static function rules(string $type, iterable $rules, bool $reset = false)
      */
     public static function pluralize(string $word) : string
     {
+        $tablized_word = self::tableize($word);
```
Comment (jwage) [inline lib/Doctrine/Common/Inflector/Inflector.php]: Can you use camelCase instead of table_case for these new variables?

**Pull request closed without merging**